entity-management-web/src/main/java/eu/europeana/entitymanagement/web/BaseRest.java

```diff
@@ -136,5 +147,35 @@ public String computeEtag(long timestamp, String format, String version){
         return DigestUtils.md5Hex(String.format("%s:%s:%s", timestamp, format, version));
     }
 
+    @SuppressWarnings("unchecked")
+	private void processLanguage(Entity entity, String languages) {
+		if(languages==null) return;
+		List<String> languagesList = Arrays.asList(languages.split(",", -1));
+    	List<Field> entityFields = EntityUtils.getAllFields(entity.getClass());
+		for(Field field : entityFields) {
+			if(field.getType().isAssignableFrom(Map.class)) {
+				Map<String, Object> currentFieldValue;
+				try {
+					currentFieldValue = (Map<String,Object>) entity.getFieldValue(field);
+				} catch (IllegalArgumentException | IllegalAccessException e) {
+					throw new FunctionalRuntimeException("An exception occured during accessing the entity field: "+field.getName(), e);
+				}
+				if(currentFieldValue==null) continue;
+				Map<String,Object> newFieldValue = new HashMap<>();
+				for(Map.Entry<String,Object> mapEntry : currentFieldValue.entrySet()) {
+					if(languagesList.contains(mapEntry.getKey())) {
```

Comment:
do not filter out URLs, which are available in the empty keys ""

Reply:
Improved.